src/ApiPort.VisualStudio/TargetPlatform.cs

```diff
@@ -89,9 +88,55 @@ public override int GetHashCode(TargetPlatform obj)
 
         public int CompareTo(TargetPlatform other)
         {
-            if (other == null) return -1;
+            if (other == null)
+            {
+                return -1;
+            }
+            else if (Equals(other))
+            {
+                return 0;
+            }
+
+            var comparedNames = string.CompareOrdinal(Name, other.Name);
+
+            if (comparedNames != 0)
+            {
+                return comparedNames;
+            }
+
+            // Adapted from:
+            // https://github.com/dotnet/corefx/blob/master/src/System.Linq/src/System/Linq/SequenceEqual.cs#L43-L55
+            using (var e1 = Versions.GetEnumerator())
```

Comment:
SequenceEquals would only tell you if your response to CompareTo should be 0. But I'd need to actually compare the items inside the list to see whether or not it was -1 or 1.